The line https://github.com/openmicroscopy/openmicroscopy/pull/5502/commits/8a7401295b5fcdc49d10ea0d721c4743127187f5#diff-d91232ef1e245a87e311c4a79a2a5f77L2529 fails with.
```
FAILED: testResetEnumerationsbyRestrictedSystemUser
java.lang.AssertionError: expected [11] but found [10]
	at org.testng.Assert.fail(Assert.java:94)
	at org.testng.Assert.failNotEquals(Assert.java:494)
	at org.testng.Assert.assertEquals(Assert.java:123)
	at org.testng.Assert.assertEquals(Assert.java:370)
	at org.testng.Assert.assertEquals(Assert.java:380)
	at integration.LightAdminRolesTest.testResetEnumerationsbyRestrictedSystemUser(LightAdminRolesTest.java:2537)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:80)
	at org.testng.internal.Invoker.invokeMethod(Invoker.java:714)
	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:901)
	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:1231)
	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:127)
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:111)
	at org.testng.TestRunner.privateRun(TestRunner.java:767)
	at org.testng.TestRunner.run(TestRunner.java:617)
	at org.testng.SuiteRunner.runTest(SuiteRunner.java:334)
	at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:329)
	at org.testng.SuiteRunner.privateRun(SuiteRunner.java:291)
	at org.testng.SuiteRunner.run(SuiteRunner.java:240)
	at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
	at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:86)
	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1198)
	at org.testng.TestNG.runSuitesLocally(TestNG.java:1123)
	at org.testng.TestNG.run(TestNG.java:1031)
	at org.testng.remote.AbstractRemoteTestNG.run(AbstractRemoteTestNG.java:126)
	at org.testng.remote.RemoteTestNG.initAndRun(RemoteTestNG.java:137)
	at org.testng.remote.RemoteTestNG.main(RemoteTestNG.java:58)
```

--- a/components/tools/OmeroJava/test/integration/LightAdminRolesTest.java
+++ b/components/tools/OmeroJava/test/integration/LightAdminRolesTest.java
@@ -28,7 +28,7 @@ import java.util.Iterator;
 import java.util.List;
 import java.util.UUID;
 
-import ome.model.acquisition.Detector;
+import ome.model.enums.DetectorType;
 import ome.model.enums.EventType;
 import omero.RLong;
 import omero.RString;
@@ -112,7 +112,7 @@ public class LightAdminRolesTest extends RolesTests {
     public void tearDown() throws Exception {
         final ITypesPrx svc = root.getSession().getTypesService();
         svc.resetEnumerations(ContrastMethod.class.getName());
-        svc.resetEnumerations(Detector.class.getName());
+        svc.resetEnumerations(DetectorType.class.getName());
         super.tearDown();
     }
 
@@ -2516,7 +2516,7 @@ public class LightAdminRolesTest extends RolesTests {
     public void testResetEnumerationsbyRestrictedSystemUser() throws Exception {
         logNewAdminWithoutPrivileges();
         final ITypesPrx types_svc = factory.getTypesService();
-        List<IObject> types = types_svc.allEnumerations(Detector.class.getName());
+        List<IObject> types = types_svc.allEnumerations(DetectorType.class.getName());
         int n = types.size();
         Iterator<IObject> i = types.iterator();
         int count = 0;
@@ -2530,11 +2530,11 @@ public class LightAdminRolesTest extends RolesTests {
         }
         //not all enum are used so we should have deleted at least one
         Assert.assertTrue(count > 0);
-        types = types_svc.allEnumerations(Detector.class.getName());
+        types = types_svc.allEnumerations(DetectorType.class.getName());
         Assert.assertEquals(types.size(), (n-count));
         //reset the deleted enumerations
-        types_svc.resetEnumerations(Detector.class.getName());
-        types = types_svc.allEnumerations(Detector.class.getName());
+        types_svc.resetEnumerations(DetectorType.class.getName());
+        types = types_svc.allEnumerations(DetectorType.class.getName());
         //We should be back to at the original list. Other enum might have been
         //added by other tests.
         Assert.assertTrue(types.size() >= n);

--- a/components/tools/OmeroJava/test/integration/LightAdminRolesTest.java
+++ b/components/tools/OmeroJava/test/integration/LightAdminRolesTest.java
@@ -28,6 +28,7 @@ import java.util.Iterator;
 import java.util.List;
 import java.util.UUID;
 
+import ome.model.acquisition.Detector;
 import ome.model.enums.EventType;
 import omero.RLong;
 import omero.RString;
@@ -111,6 +112,7 @@ public class LightAdminRolesTest extends RolesTests {
     public void tearDown() throws Exception {
         final ITypesPrx svc = root.getSession().getTypesService();
         svc.resetEnumerations(ContrastMethod.class.getName());
+        svc.resetEnumerations(Detector.class.getName());
         super.tearDown();
     }
 
@@ -2514,7 +2516,7 @@ public class LightAdminRolesTest extends RolesTests {
     public void testResetEnumerationsbyRestrictedSystemUser() throws Exception {
         logNewAdminWithoutPrivileges();
         final ITypesPrx types_svc = factory.getTypesService();
-        List<IObject> types = types_svc.allEnumerations(ContrastMethod.class.getName());
+        List<IObject> types = types_svc.allEnumerations(Detector.class.getName());
         int n = types.size();
         Iterator<IObject> i = types.iterator();
         int count = 0;
@@ -2528,11 +2530,11 @@ public class LightAdminRolesTest extends RolesTests {
         }
         //not all enum are used so we should have deleted at least one
         Assert.assertTrue(count > 0);
-        types = types_svc.allEnumerations(ContrastMethod.class.getName());
+        types = types_svc.allEnumerations(Detector.class.getName());
         Assert.assertEquals(types.size(), (n-count));
         //reset the deleted enumerations
-        types_svc.resetEnumerations(ContrastMethod.class.getName());
-        types = types_svc.allEnumerations(ContrastMethod.class.getName());
+        types_svc.resetEnumerations(Detector.class.getName());
+        types = types_svc.allEnumerations(Detector.class.getName());
         //We should be back to at the original list. Other enum might have been
         //added by other tests.
         Assert.assertTrue(types.size() >= n);

--- a/components/tools/OmeroJava/test/integration/LightAdminRolesTest.java
+++ b/components/tools/OmeroJava/test/integration/LightAdminRolesTest.java
@@ -2533,8 +2533,9 @@ public class LightAdminRolesTest extends RolesTests {
         //reset the deleted enumerations
         types_svc.resetEnumerations(ContrastMethod.class.getName());
         types = types_svc.allEnumerations(ContrastMethod.class.getName());
-        //We should be back to the original list
-        Assert.assertEquals(types.size(), n);
+        //We should be back to at the original list. Other enum might have been
+        //added by other tests.
+        Assert.assertTrue(types.size() >= n);
     }
 
     /**

